[jiancheng37] iP

jiancheng37 opened the pull request:
# Jimmy
Jimmy frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ Super fast to use

All you need to do is,
1. download it from [here] (https://github.com/jiancheng37/ip)
2. double-click it
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
def hello_world():
    print("Hello, World!")

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jianchenglow committed:
Changed chatbot name to Jimmy and let it greet the user accordingly

jianchenglow committed:
Echos user commands and exits once user types bye

jianchenglow committed:
Fixed a bug where bye shows before exiting the program

jianchenglow committed:
Gave my bot the ability to store user's text and display them when prompted

jianchenglow committed:
Add the ability to mark and unmark tasks

jianchenglow committed:
Added support for Todo, Events and Deadlines

jianchenglow committed:
Added testing capability using I/O redirection technique

jianchenglow committed:
Added error handling

jianchenglow committed:
Added support for deleting tasks from the list

jianchenglow committed:
Implement saving feature

jianchenglow committed:
Merge Level 7 implementation

jianchenglow committed:
Implement Date and Time understanding

jianchenglow committed:
Merge Level 8 implementation

jianchenglow committed:
Implement OOP in file architecture

jianchenglow committed:
Merge A-MoreOOP implementation

jianchenglow committed:
Organize files into package

jianchenglow committed:
Merge add-gradle-support into master

jianchenglow committed:
Implement JUnit test for Task and TaskList

jianchenglow committed:
Package the app as an executable JAR file

jianchenglow committed:
Add JavaDoc comments to all classes and methods

jianchenglow committed:
Adhere to coding standards

jianchenglow committed:
Add find functionality

jianchenglow committed:
Merge coding standard improvements

jianchenglow committed:
Merge Level 9 implementation

mingshan2705 commented on src/main/java/jimmy/Parser.java:
```diff
@@ -0,0 +1,80 @@
+        String commandWord = words[0];
+
+        switch (commandWord) {
+        case "bye":
```
I like that you aligned the `case` with `switch`, following the Java Coding Standard.

mingshan2705 commented on src/main/java/jimmy/FindCommand.java:
```diff
@@ -0,0 +1,27 @@
+        this.input = input;
+    }
+
+    @Override
```
Perhaps you could include documentation for this package, similar to what you did for the other Commands

jianchenglow committed:
Implement CheckStyle

jianchenglow committed:
Implement Gui

jianchenglow committed:
Merge CheckStyle implementation

jianchenglow committed:
Merge Level 10 implementation

yuyang011 commented on src/main/java/jimmy/Storage.java:
```diff
@@ -0,0 +1,121 @@
+package jimmy;
+
+import java.io.BufferedReader;
```
I like how u ordered the import statements.

yuyang011 commented on src/main/java/jimmy/Storage.java:
```diff
@@ -0,0 +1,121 @@
+    public ArrayList<Task> load() throws JimmyException {
+        ArrayList<Task> tasks = new ArrayList<>();
+
+        try (BufferedReader br = new BufferedReader(new FileReader(filePath.toString()))) {
```
Should br be used instead of bufferReader as it could be too short and confusing if it appears again later in other lines for the reader.

yuyang011 commented on src/main/java/jimmy/Storage.java:
```diff
@@ -0,0 +1,121 @@
+     * @throws JimmyException if an I/O error occurs during file writing.
+     */
+    public void save(ArrayList<Task> tasks) throws JimmyException {
+        try (BufferedWriter bw = new BufferedWriter(new FileWriter(filePath.toString()))) {
```
Should bw be used instead of bufferwritter as it could be too short and confusing if it appears later in other lines for the reader.

yuyang011 commented on src/main/java/jimmy/Storage.java:
```diff
@@ -0,0 +1,121 @@
+     * Ensures the file and its parent directories exist.
+     * Creates them if they do not exist.
+     */
+    private void ensureFileExists() {
```
I like how you used the name to explain what the function does.

yuyang011 commented on src/main/java/jimmy/TaskList.java:
```diff
@@ -0,0 +1,73 @@
+     *
+     * @return the list of all tasks.
+     */
+    public ArrayList<Task> getAllTasks() {
+        return tasks;
```
I like how you clearly distinguished multi-valued variables.

yuyang011 reviewed (commented):
LGTM, mostly follow the naming guidelines, only some parts to fix

jianchenglow committed:
Improve code robustness in Parser, Storage, and MarkCommand

Parser, Storage, and MarkCommand contain logic that assumes certain
conditions hold true but do not enforce these assumptions explicitly.

Without proper validation, unexpected inputs or incorrect state
assumptions may lead to runtime errors that are difficult to debug.

Let's add `assert` statements to document and enforce these assumptions:

* `Parser.java`: Ensure input commands are non-null and non-empty before
  processing.
* `Storage.java`: Verify that the provided file path is valid and that
  the storage file exists after initialization.
* `MarkCommand.java`: Prevent negative task indices and ensure the task
  list is not null before execution.

These assertions act as an additional safeguard during development
and help detect logical errors early.

jianchenglow committed:
Refactor Parser and Storage for improved readability and maintainability

Parser and Storage contained repetitive logic, deep nesting, and magic strings,
which made the code harder to maintain and read. Additionally, error handling
was inconsistent, and some assumptions were not enforced.

To improve code quality, let's:
- Extract duplicate validation logic in `Parser` into helper methods
  (`validateNoArguments()` and `validateHasArguments()`).
- Replace magic strings in `Storage` with named constants for better maintainability.
- Use early returns to reduce deep nesting and improve readability.
- Extract task parsing logic in `Storage` into dedicated methods (`parseTask()`,
  `parseDeadline()`, `parseEvent()`).
- Add assertions to enforce assumptions, such as ensuring files exist before
  reading and validating non-null task lists before saving.
- Improve exception messages to be more informative, helping users understand errors.

These changes enhance the overall structure, maintainability, and robustness of
the code while following clean code principles.

jiancheng37 committed:
Merge pull request #1 from jiancheng37/branch-A-Assertions

Improve code robustness in Parser, Storage, and MarkCommand

jianchenglow committed:
Merge branch 'master' into branch-A-CodeQuality

jiancheng37 committed:
Merge pull request #2 from jiancheng37/branch-A-CodeQuality

Implement code quality changes

jianchenglow committed:
Implement friendlier syntax for commands

jiancheng37 committed:
Merge branch 'master' into branch-C-FriendlerSyntax

jiancheng37 committed:
Merge pull request #3 from jiancheng37/branch-C-FriendlerSyntax

Implement friendlier syntax for commands

jianchenglow committed:
Integrate profile pictures into GUI

jianchenglow committed:
Merge A-BetterGui implementation

jianchenglow committed:
Add User Guide to README and set up Github Pages

jianchenglow committed:
Merge A-UserGuide implementation
